src/parser/ir/ir_parser.cc

```diff
@@ -98,34 +100,51 @@ const Operation& IrProgramParser::ConstructOperation(
                   const std::string& value_id =
                       CHECK_NOTNULL(named_value_context)->VALUE_ID()->getText();
                   // Operation Result Value
-                  auto find_value_id_operation = operation_map_.find(value_id);
-                  // check in place as we do not support operation lists yet.
-                  CHECK(find_value_id_operation == operation_map_.end());
-                  if (find_value_id_operation != operation_map_.end()) {
-                    return Value(value::OperationResult(
-                        *find_value_id_operation->second, value_id));
+                  auto find_value_id_operation = value_map_.find(value_id);
+                  if (find_value_id_operation != value_map_.end()) {
```

Comment:
Actually change this to a check and get rid of the storage case altogether. We plan to deprecate storage anyway. 

(Do the tests have coverage for storage?)

Also, add a death test for the case where a value will not be found.